### PR TITLE
chore: Remove deprecations scheduled for 4.0.0 release

### PR DESCRIPTION
**Pull request by gziolo**
## Description
This PR removed all deprecations schedule to be cleaned up as part of 4.0.0 release:

- `wp.editor.RichTextProvider` has been removed. Please use `wp.data.select( 'core/editor' )` methods instead.
- `wp.components.Draggable` as a DOM node drag handler has been removed. Please, use `wp.components.Draggable` as a wrap component for your DOM node drag handler.
- `wp.i18n.getI18n` has been removed. Use `__`, `_x`, `_n`, or `_nx` instead.
- `wp.i18n.dcnpgettext` has been removed. Use `__`, `_x`, `_n`, or `_nx` instead.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by nosolosw** [comment]
Checked `draggable` removal: docs are ok (general + components package), there is no code related to the deprecation left (JS or CSS), and every part of Gutenberg is using the API released in 3.8.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

File: packages/i18n/CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
-## 2.1.0 (Unreleased)
+## 3.0.0 (Unreleased)
 
-### Deprecations
+### Braking Change
```
Comment (aduth) [inline packages/i18n/CHANGELOG.md]: Typo: breaking

File: packages/i18n/CHANGELOG.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 
-- `getI18n` has been deprecated. Use `__`, `_x`, `_n`, or `_nx` instead.
-- `dcnpgettext` has been deprecated. Use `__`, `_x`, `_n`, or `_nx` instead.
+- `wp.i18n.getI18n` has been removed. Use `__`, `_x`, `_n`, or `_nx` instead.
```
Comment (aduth) [inline packages/i18n/CHANGELOG.md]: Why did we at the global reference here? They aren't relevant for the package.

File: packages/i18n/src/index.js [hunk 3]
```diff
@@ -4,11 +4,6 @@
 import Jed from 'jed';
 import memoize from 'memize';
 
-/**
- * WordPress dependencies
- */
-import deprecated from '@wordpress/deprecated';
```
Comment (aduth) [inline packages/i18n/src/index.js]: This package no longer depends on deprecated, so it should be removed from the `package.json` file and from `client-assets.php`.

**Pull request merged**